libr/debug/p/native/linux/linux_debug.c

```diff
@@ -517,3 +521,1439 @@ RList *linux_desc_list (int pid) {
+	char paux[ELF_PRARGSZ];
+	char *p;
+	int i;
+	int bytes_left;
+
+	p = r_mem_dup (pfname, size_psargs);
+	if (!p) return NULL;
+
+	bytes_left = strlen (pfname);
+	buffer = strchr (buffer, '\0');
+
+	for(i = 0; i + bytes_left < len && i + bytes_left < (size_psargs - 1); i++) {
+		if (buffer[i] == '\0') buffer[i] = ' ';
+		paux[i] = buffer[i];
+	}
+
+	paux[i] = '\0';
+	strncat (p, paux, size_psargs - bytes_left - 1);
+
+	return p;
+}
+
+static void debug_print_prpsinfo(prpsinfo_t *p) {
+	eprintf ("prpsinfo.pr_state: %d\n", p->pr_state);
+	eprintf ("prpsinfo.pr_sname: %c\n", p->pr_sname);
+	eprintf ("prpsinfo.pr_zomb: %d\n", p->pr_zomb);
+	eprintf ("prpsinfo.pr_nice: %d\n", p->pr_nice);
+	eprintf ("prpsinfo.pr_flags: %ld\n", p->pr_flag);
+	eprintf ("prpsinfo.pr_uid: %d\n", p->pr_uid);
+	eprintf ("prpsinfo.pr_gid: %d\n", p->pr_gid);
+	eprintf ("prpsinfo.pr_pid: %d\n", p->pr_pid);
+	eprintf ("prpsinfo.pr_ppid: %d\n", p->pr_ppid);
+	eprintf ("prpsinfo.pr_pgrp: %d\n", p->pr_pgrp);
+	eprintf ("prpsinfo.pr_sid: %d\n", p->pr_sid);
+	eprintf ("prpsinfo.pr_fname: %s\n", p->pr_fname);
+	eprintf ("prpsinfo.pr_psargs: %s\n", p->pr_psargs);
+}
+
+static prpsinfo_t *linux_get_prpsinfo(RDebug *dbg, proc_stat_content_t *proc_data) {
+	char file[128];
+	int len;
+	pid_t mypid;
+	char *buffer;
+	char *pfname;
+	char *ppsargs;
+	char *basename; /* pr_fname stores just the exec, withouth the path */
+	const char prog_states[] = "RSDTZW";				/* fs/binfmt_elf.c from kernel */
+	prpsinfo_t *p;
+
+	pfname = ppsargs = basename = NULL;
+
+	p = R_NEW0 (prpsinfo_t);
+	if (!p) {
+		eprintf ("Couldn't allocate memory for prpsinfo_t\n");
+		return NULL;
+	}
+
+	p->pr_pid = mypid = dbg->pid;
+
+	/* Start filling pr_fname and pr_psargs */
+	snprintf (file, sizeof(file), "/proc/%d/cmdline", mypid);
+	eprintf ("r_file_slurp\n");
+	buffer = r_file_slurp (file, &len);
+	if (!buffer) {
+		eprintf ("buffer NULL\n");
+		return NULL;
+	}
+	
+	buffer = r_file_slurp (file, &len);
+	pfname = prpsinfo_get_fname (buffer);	
+	if (!pfname) {
+		eprintf ("prpsinfo_get_fname: couldn't allocate memory\n");
+		goto error;
+	}
+	
+	basename = get_basename (pfname, strlen(pfname));
+	strncpy (p->pr_fname, basename, sizeof(p->pr_fname));
+
+	ppsargs = prpsinfo_get_psargs (buffer, pfname, sizeof(p->pr_psargs), len);
+	if (!ppsargs) {
+		eprintf ("prpsinfo_get_psargs: couldn't allocate memory\n");
+		return NULL;
+	}
+
+	strncpy (p->pr_psargs, ppsargs, sizeof(p->pr_psargs));
+	free (buffer);
+	free (ppsargs);
+	free (pfname);
+	
+	p->pr_sname = proc_data->s_name;
+	p->pr_zomb = (p->pr_sname == 'Z') ? 1 : 0;
+	p->pr_state = strchr (prog_states, p->pr_sname) - prog_states;
+	p->pr_ppid = proc_data->ppid;
+	p->pr_pgrp = proc_data->pgrp;
+	p->pr_sid = proc_data->sid;
+	p->pr_flag = proc_data->flag;
+	p->pr_nice = proc_data->nice;
+	p->pr_uid = proc_data->uid;
+	p->pr_gid = proc_data->gid;
+	
+	debug_print_prpsinfo(p);
+	eprintf ("linux_get_prpsinfo: end\n");
+
```

Comment:
If you're refering to the second call to

`buffer = r_file_slurp (file, &len);`

yes, otherwise I can't see it